servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ToStH1ClientDuplexHandler.java

```diff
@@ -56,15 +57,17 @@
     @Nullable
     private HttpRequestMethod method;
 
-    H2ToStH1ClientDuplexHandler(boolean sslEnabled, BufferAllocator allocator, HttpHeadersFactory headersFactory) {
-        super(headersFactory);
+    H2ToStH1ClientDuplexHandler(boolean sslEnabled, BufferAllocator allocator, HttpHeadersFactory headersFactory,
+                                CloseHandler closeHandler) {
+        super(headersFactory, closeHandler);
         this.scheme = sslEnabled ? HttpScheme.HTTPS : HttpScheme.HTTP;
         this.allocator = allocator;
     }
 
     @Override
     public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) {
         if (msg instanceof HttpRequestMetaData) {
+            closeHandler.protocolPayloadBeginOutbound(ctx);
```

Comment:
We are using `UnsupportedProtocolHandler` which does not implement these callbacks so we are not really verifying whether these callbacks are correct. Should we hold off adding calls to `CloseHandler` other than `protocolPayloadEndOutbound()` till we need them?

Reply:
I wanted to make it more consistent with H1, but I'm OK with leaving them out.

Reply:
Ok Ya either we validate that we are doing the rite thing or leave it out. I would prefer leaving them out.